airbyte-integrations/connectors/source-mailersend/source_mailersend/mailersend.yaml

```diff
@@ -0,0 +1,50 @@
+version: "0.1.0"
+
+definitions:
+  selector:
+    extractor:
+      field_pointer: ["data"]
+  requester:
+    url_base: "https://api.mailersend.com/v1"
+    http_method: "GET"
+    authenticator:
+      type: BearerAuthenticator
+      api_token: "{{ config['api_token'] }}"
+    request_options_provider:
+      request_parameters:
+        limit: "100"
+        date_from: "{{ config['date_from'] }}"
+        date_to: "{{ config['date_to'] }}"
+  retriever:
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: "DefaultPaginator"
+      page_size_option:
+        inject_into: "request_parameter"
+        field_name: "page_size"
+      pagination_strategy:
+        type: "PageIncrement"
+        page_size: 100
+      page_token_option:
+        inject_into: "request_parameter"
+        field_name: "page"
+      url_base: "*ref(definitions.requester.url_base)"
+    requester:
+      $ref: "*ref(definitions.requester)"
+  base_stream:
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+  activity_stream:
+    $ref: "*ref(definitions.base_stream)"
+    $options:
+      name: "activity"
+      primary_key: "id"
+      path: "/activity/{{ config['domain_id'] }}"
```

Comment:
I saw the API supports the query parameter from/to using dates, this allow using incremental syncs. @cirdes what about implementing this feature for this connector? I can help you if you need some help.

Reply:
@marcosmarxm, the API does supports query parameter using dates. But it expect the params to be a integer like 1443651141 (utc in seconds). DatetimeStreamSlicer is using date as strings "2021-02-01T00:00:00.000000+0000". 

Is it possible to use DatetimeStreamSlicer in this case?

Reply:
@girarda can you have me if it is possible to convert to timestamp?

Reply:
thanks guy, I will have a look on it!

Reply:
I don't know if a got the point here.

The challenge is not about the stream_slicer itself. 
In the spec.yml I'm asking the user to enter start_time(or date_from) as a number because I need to send start_time as number to Mailersend's API.

mailersend.yml
```
request_options_provider:
  request_parameters:
    limit: "100"
    start_time: "{{ config['date_from'] }}"
```

But if a change start_time to be a string link "2022-11-05" I will need to cast the string into Date and then get the unix timestamp but its seems to be not possible in the yml.
In sendgrid exemple it is doing that using python and pendulum lib -> https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid/source_sendgrid/streams.py#L98

It's seems to be a complex scenario or I'm missing a simple solution here?

cc: @marcosmarxm

Reply:
@cirdes, you can configure the datetime format of the stream slicer's fields. The `%s` directive converts the datetime to a timestamp.

Using Sendgrid as an example:
```
  stream_slicer:
    type: "DatetimeStreamSlicer"
    start_datetime:
      datetime: "{{ config['start_time'] }}"
      datetime_format: "%s"
```

https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-sendgrid/source_sendgrid/sendgrid.yaml#L47-L63

Reply:
@cirdes let's schedule a time tomorrow to implement the stream slicer.